images/ipfailover/keepalived/lib/config-generators.sh

```diff
@@ -212,17 +183,29 @@ function generate_vrrpd_instance_config() {
   local priority=${5:-"10"}
   local instancetype=${6:-"slave"}
 
-  local vipname=$(scrub "$1")
+  local vipname ; vipname=$(scrub "$1")
   local initialstate=""
-  local preempt=${PREEMPTION:-"$DEFAULT_PREEMPTION_STRATEGY"}
+  local preempt=${PREEMPTION:-"${DEFAULT_PREEMPTION_STRATEGY}"}
   local vrrpidoffset=${HA_VRRP_ID_OFFSET:-0}
 
-  [ "$instancetype" = "master" ] && initialstate="state MASTER"
+  [ "${instancetype}" = "master" ] && initialstate="state MASTER"
 
-  local instance_name=$(generate_vrrp_instance_name "$servicename" "$iid")
+  local instance_name ; instance_name=$(generate_vrrp_instance_name "${servicename}" "${iid}")
 
-  local auth_section=$(generate_authentication_info "$servicename")
-  local vip_section=$(generate_vip_section "$vips" "$interface")
+  local auth_section ; auth_section=$(generate_authentication_info "${servicename}")
+  local vip_section ; vip_section=$(generate_vip_section "${vips}" "${interface}")
+  # Emit sync_group
+  local prefix ; prefix="$(vrrp_instance_basename "$1")"
+  echo ""
+  echo "vrrp_sync_group group_${instance_name} {"
+  echo "   group {"
+  echo "      ${instance_name}   # VIP ${vips}"
+  echo "   }"
+  if [[ -n $HA_NOTIFY_SCRIPT ]]; then
+      echo "   notify \"${HA_NOTIFY_SCRIPT}\""
+  fi
+  echo "}"
```

Comment:
@ramr are you certain that the vrrp_sync_group is optional? I have not found that in the docs yet.